### ISSUE TITLE
[js] Unbound variable 

### ISSUE THREAD

**Issue by AlexHaxe** [HaxeFoundation/haxe#8791]
After fixing some EFunction changes in checkstyle, running `haxe buildJS.hxml` with Haxe4 rc4 (and latest git) in checkstyle project shows these two messages:
```
Unbound variable, please report this
{
	v_id = 30746;
	v_name = binary;
	v_type = Bool;
	v_capture = false;
	v_extra = None;
	v_meta = [];
}
Unbound variable, please report this
{
	v_id = 30792;
	v_name = binary;
	v_type = Bool;
	v_capture = false;
	v_extra = None;
	v_meta = [];
}
```

**Comment by Simn** [comment]
This is embarrassingly easy to reproduce:

```haxe
class Main {
	static public function main() {
		var a = write(true);
	}

	static inline function write(binary = true) {
		return "";
	}
}
```

**Comment by RealyUniqueName** [comment]
Open until merged into 4.0_bugfix
